File: services/booking/src/main/scala/com/wedriveu/services/booking/util/Dates.scala
package com.wedriveu.services.booking.util

import java.text.SimpleDateFormat
import java.util.Date

/** Simple date-utility object
  *
  * @author Nicola Lasagni on 18/08/2017.
  */
object Dates {

  private val Format = "dd/MM/yyyy HH:mm"
  private val DateFormat = new SimpleDateFormat(Format)

  /** Formats a date into a predefined [[Format]]
    *
    * @param date The date to be formatted.
    * @return A [[String]] representation of the date provided.
    */
  def format(date: Date): String = {
    if (date == null) "" else DateFormat.format(date)
  }

}

File: mobile/src/main/java/com/wedriveu/mobile/util/Dates.java
package com.wedriveu.mobile.util;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/**
 * Utility class for manipulating dates.
 *
 * @author Nicola Lasagni on 26/08/2017.
 */
public class Dates {

    private static final String FORMAT = "dd/MM/yyyy HH:mm";
    private static final DateFormat DATE_FORMAT = new SimpleDateFormat(FORMAT, Locale.getDefault());

    /**
     * Formats a date by using {@linkplain #FORMAT}.
     *
     * @param date the date to be formatted
     * @return the formatted date
     */
    public static String format(Date date) {
        if (date == null) {
            return "";
        }
        return DATE_FORMAT.format(date);
    }

}
